libs/geo/src/main/java/org/elasticsearch/geometry/utils/WellKnownBinary.java

```diff
@@ -217,6 +217,19 @@ public static Geometry fromWKB(GeometryValidator validator, boolean coerce, byte
         return geometry;
     }
 
+    /**
+     * Reads a point coordinate from the given WKB byte array with offset.
+     */
+    public static double coordinateFromWKB(int coordinateIndex, byte[] wkb, int offset, int length) {
```

Comment:
-1 to add such a method here sorry

Reply:
I think you can just move it as a private method where you need it.

Reply:
OK. I'll remove that. I added based on Nik's comment, but he did not strictly need it, just an idea to reduce object creation.

Reply:
You can add it but not in the library. Let's keep it simple this library, read from / write to wkb. If you want to add that optimisation, add it in the ESQL module where you need it,